common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -230,6 +240,70 @@ protected final void setAuthorizationEndpoint(final String authorizationEndpoint
         mAuthorizationEndpoint = authorizationEndpoint;
     }
 
+    public AuthorizationResult getDeviceCode(@NonNull final MicrosoftStsAuthorizationRequest authorizationRequest) throws IOException {
+        final String methodName = ":getDeviceCode";
+
+        final String urlBody = ((MicrosoftStsOAuth2Configuration) mConfig).getAuthorityUrl().toString() + "/oauth2/v2.0/devicecode";
+
+        // Set up headers and request body
+        final String requestBody = ObjectMapper.serializeObjectToFormUrlEncoded(authorizationRequest);
+        final Map<String, String> headers = new TreeMap<>();
+        headers.put(CLIENT_REQUEST_ID, DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
+        headers.putAll(EstsTelemetry.getInstance().getTelemetryHeaders());
+
+        // Send request
+        final HttpResponse response = HttpRequest.sendPost(
+                new URL(urlBody),
+                headers,
+                requestBody.getBytes(ObjectMapper.ENCODING_SCHEME),
+                DEVICE_CODE_CONTENT_TYPE
+        );
+
+        // Create the authorization result
+        // MicrosoftSTAuthorizationResultFactory not used since no Intent is being created
+        AuthorizationResult authorizationResult;
+
+        // Check if the request was successful
+        // Any code below 300 (HTTP_MULT_CHOICE) is considered a success
+        if (response.getStatusCode() < HttpsURLConnection.HTTP_MULT_CHOICE){
+            // Get and parse response body
+            final HashMap<String, String> parsedResponseBody = new Gson().fromJson(response.getBody(), new TypeToken<HashMap<String, String>>() {}.getType());
+
+            // Create response and result objects
+            // Not sure where "code" is stored. Not found in the input stream...
+            final MicrosoftStsAuthorizationResponse authorizationResponse =
+                    new MicrosoftStsAuthorizationResponse(null, authorizationRequest.getState(), parsedResponseBody);
+            authorizationResult = new MicrosoftStsAuthorizationResult(AuthorizationStatus.SUCCESS, authorizationResponse);
+
+            Logger.verbose(
+                    TAG + methodName,
+                    "Device Code Flow authorization successful..."
+            );
+        }
```

Comment:
nit: early return? (so there's no need to create a AuthorizationResult pointer in line 264).